.github/workflows/ci-comment.yml

```diff
@@ -1,58 +1,32 @@
     if: ${{ github.event.workflow_run.conclusion == 'failure' }}
     steps:
-      - name: 'Download artifact'
-        uses: actions/github-script@v6
+      - name: Fetch PR Data
+        uses: dawidd6/action-download-artifact@6765a42d86407a3d532749069ac03705ad82ebc6
         with:
-          script: |
-            let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
-               owner: context.repo.owner,
-               repo: context.repo.repo,
-               run_id: context.payload.workflow_run.id,
-            });
-            let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
-              return artifact.name == "pr_number"
-            })[0];
-            let download = await github.rest.actions.downloadArtifact({
-               owner: context.repo.owner,
-               repo: context.repo.repo,
-               artifact_id: matchArtifact.id,
-               archive_format: 'zip',
-            });
-            let fs = require('fs');
-            fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/pr_number.zip`, Buffer.from(download.data));
-
-      - name: 'Unzip artifact'
-        run: unzip pr_number.zip
+          name: pr_number
+          run_id: github.event.workflow_run.id
 
-      - name: 'Comment on PR'
-        uses: actions/github-script@v6
+      - name: Save PR Data
+        id: save-pr-data
+        run: |
+          echo "::set-output name=issue_number::$(cat pr_number)"
+          echo "::set-output name=pr_sha::$(cat pr_sha)"
+          echo "::set-output name=merge_sha::$(cat merge_sha)"
+      
+      - name: Add Comment
+        uses: marocchino/sticky-pull-request-comment@39c5b5dc7717447d0cba270cd115037d32d28443
         with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
-          script: |
-            let fs = require('fs');
-            let issue_number = Number(fs.readFileSync('./pr_number', 'utf8').trim());
-            let pr_sha = fs.readFileSync('./pr_sha', 'utf8').trim();
-            let merge_sha = fs.readFileSync('./merge_sha', 'utf8').trim();
-            let pull = await github.rest.pulls.get({
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              pull_number: issue_number
-            });
-            await github.rest.issues.createComment({
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              issue_number: issue_number,
-              body: `The commit ${pr_sha} (as a parent of ${merge_sha}) contains errors. Please inspect the [Run Summary](${context.payload.workflow_run.html_url}) for details.`
-            });
-
+          recreate: true
+          message: |
+            The commit ${{ steps.save-pr-data.outputs.pr_sha }} (as a parent of ${{ steps.save-pr-data.outputs.merge_sha }}) contains errors.
+            Please inspect the [Run Summary](https://github.com/ethereum/EIPs/pull/${{ steps.save-pr-data.outputs.pr_number }}/files) for details.
```

Comment:
This only links to the files tab, which won't show all the errors if the errors aren't close enough to a changed line.

Reply:
This isn't true anymore. In the past few days, GitHub implemented viewing errors on unchanged lines and even unchanged files.